GSA-TTS/FAC
# Search Results - Finding my ALN and Finding all ALN

# Search Results - Finding my ALN and Finding all ALN

Issue: https://github.com/GSA-TTS/FAC/issues/2918
Epic: https://github.com/GSA-TTS/FAC/issues/2652

## Changes:
Searching on any ALN results in two new columns in the results table. The goal is to see if a record has any findings, and if there are, if they are under my searched-for ALNs. So, there is now:

1. "Finding my ALN". If the record has an award with a finding under a searched for ALN, this column is 'Y'.
2. "Finding all ALN". If the record has an award with a finding NOT under a searched-for ALN, this column is 'Y'. 

These columns are not mutually exclusive for a record. If a record has awards under several programs (as many do), then they can have findings under multiple ALNs. Then, it's possible to see both columns as 'Y'. 

Minor changes:
1. ALN test cases. Includes tests for these new columns.
2. The results table is scrollable, and slightly denser.
3. The little button hiding in the bottom left of the search screen is gone.
4. Date range end date auto-populates correctly.
5. A link to the search instructions page appears before making a search.

## Examples:
A record has no findings. When it appears in an ALN search, both "Finding my ALN" and "Finding all ALN" columns will be 'N'.

A record has one award under agency '00', and it has a finding. When we search for ALN '00', "Finding my ALN" will be 'Y'. 

A record has awards under both agencies '00' and '99'. One award under agency '99' has a finding. When we search for ALN '00', "Finding all ALN" will be 'Y'.

A record has awards under both agencies '00' and '99'. There are findings for awards in both agencies. When we search for ALN '00', both "Finding my ALN" and "Finding all ALN" will be 'Y'.

## How to Test:
Refer to the above examples to help find (or simulate, if you're crazy) test cases.
1. Switch to this branch and run normally.
2. Get some data into dissemination. 
    1. This can be one with something akin to:
        
        ```
        python manage.py end_to_end_test_data_generator --email <signed in user's email here> --dbkeys 
        "101078,101101,101352,101382,101546,101579,101590,101590,101893,10213,102138,102184" --years 
        "22,22,22,22,22,22,22,22,22,22,22,22"
        ```
    2. This can also be done with Cypress, but diversity will be lacking.
    3. This can also be done manually, but don't do that unless you have a lot of saved up workbooks and time to spare.
6. Make some ALN searches to confirm the new columns behave properly. 

If you use the above code snippet, there are examples of records that meet all the example scenarios. 

## Screenshot:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/61d4c6b1-ace8-4f22-8a50-2d90102f9c6d)

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
